appcues/src/main/java/com/appcues/debugger/DebuggerViewModel.kt

```diff
@@ -107,36 +109,59 @@ internal class DebuggerViewModel(
         }
     }
 
-    fun onStart(deepLinkPath: String?) {
-        if (deepLinkPath.isNullOrEmpty()) {
-            // if no path is given, and currently expanded, go back to idle
-            if (_uiState.value is Expanded) {
-                _uiState.value = Idle
+    fun onStart(mode: DebugMode, reset: Boolean) {
+        this.mode = mode
+
+        when (mode) {
+            is Debugger -> {
+                val deepLinkPath = mode.deepLinkPath
+                if (deepLinkPath.isNullOrEmpty()) {
+                    // if no path is given, and currently expanded, go back to Idle
+                    if (_uiState.value is Expanded ||
+                        // OR if showing FAB but the mode changed, reset to new Idle mode
+                        (_uiState.value is Idle && reset)) {
```

Comment:
this is a spot where that `reset` is used - going from one type of Idle to another.